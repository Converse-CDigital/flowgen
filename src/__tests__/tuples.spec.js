// @flow

import { compiler, beautify } from "..";

it("should handle tuples", () => {
  const ts = `
  type T1 = [number, string?];
  type T2 = [number, ...string[]];
  `;
  const result = compiler.compileDefinitionString(ts);
  expect(beautify(result)).toMatchSnapshot();
});
