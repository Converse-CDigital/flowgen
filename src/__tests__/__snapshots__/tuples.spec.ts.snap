// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle tuples 1`] = `
"declare type T1 = [number, string | void];
declare type T2 = [number] & string[];
"
`;
